commit c70483584bf0275728e43e3a49bb669779ec30e5
Author: Ryan Tandy <rtandy@sd63.bc.ca>
Date:   Wed Sep 24 10:16:06 2014 -0700

    Fix root's reboot command

diff --git a/ica/src/LocalSystemIca.cpp b/ica/src/LocalSystemIca.cpp
index cf84236..1079cd0 100644
--- a/ica/src/LocalSystemIca.cpp
+++ b/ica/src/LocalSystemIca.cpp
@@ -173,7 +173,7 @@ void reboot( void )
 #else
 	if( LocalSystem::User::loggedOnUser().name() == "root" )
 	{
-		QProcess::startDetached( "poweroff" );
+		QProcess::startDetached( "reboot" );
 	}
 	else
 	{
