diff -Naur italc-2.0.0.orig/lib/include/ItalcCoreConnection.h italc-2.0.0/lib/include/ItalcCoreConnection.h
--- italc-2.0.0.orig/lib/include/ItalcCoreConnection.h	2011-08-03 23:18:00.000000000 +0200
+++ italc-2.0.0/lib/include/ItalcCoreConnection.h	2012-03-21 09:01:14.000000000 +0100
@@ -106,6 +106,9 @@
 
 	void reportSlaveStateFlags();
 
+signals:
+	void receivedUserInfo( const QString &, const QString & );
+	void receivedSlaveStateFlags( const int );
 
 private slots:
 	void initNewClient( rfbClient *client );
diff -Naur italc-2.0.0.orig/lib/include/ItalcVncConnection.h italc-2.0.0/lib/include/ItalcVncConnection.h
--- italc-2.0.0.orig/lib/include/ItalcVncConnection.h	2011-08-03 23:18:00.000000000 +0200
+++ italc-2.0.0/lib/include/ItalcVncConnection.h	2012-03-21 09:02:45.000000000 +0100
@@ -190,6 +190,7 @@
 	void passwordRequest();
 	void outputErrorMessage( const QString &message );
 	void connected();
+	void stateChanged( const int );
 
 
 public slots:
diff -Naur italc-2.0.0.orig/lib/src/ItalcCoreConnection.cpp italc-2.0.0/lib/src/ItalcCoreConnection.cpp
--- italc-2.0.0.orig/lib/src/ItalcCoreConnection.cpp	2011-08-03 23:18:00.000000000 +0200
+++ italc-2.0.0/lib/src/ItalcCoreConnection.cpp	2012-03-21 09:01:14.000000000 +0100
@@ -132,10 +132,12 @@
 		{
 			m_user = m.arg( "username" );
 			m_userHomeDir = m.arg( "homedir" );
+			emit receivedUserInfo( m_user, m_userHomeDir );
 		}
 		else if( m.cmd() == ItalcCore::ReportSlaveStateFlags )
 		{
 			m_slaveStateFlags = m.arg( "slavestateflags" ).toInt();
+			emit receivedSlaveStateFlags( m_slaveStateFlags );
 		}
 		// TODO: plugin hook
 		else
diff -Naur italc-2.0.0.orig/lib/src/ItalcVncConnection.cpp italc-2.0.0/lib/src/ItalcVncConnection.cpp
--- italc-2.0.0.orig/lib/src/ItalcVncConnection.cpp	2011-08-03 23:18:00.000000000 +0200
+++ italc-2.0.0/lib/src/ItalcVncConnection.cpp	2012-03-21 09:04:43.000000000 +0100
@@ -513,6 +513,7 @@
 void ItalcVncConnection::run()
 {
 	m_state = Disconnected;
+	emit stateChanged( m_state );
 	m_stopped = false;
 
 	rfbClientLog = hookOutputHandler;
@@ -569,6 +570,7 @@
 			emit connected();
 
 			m_state = Connected;
+			emit stateChanged( m_state );
 			if( m_framebufferUpdateInterval < 0 )
 			{
 				rfbClientSetClientData( m_cl, (void *) 0x555, (void *) 1 );
@@ -581,15 +583,18 @@
 			if( argc < 0 )
 			{
 				m_state = HostUnreachable;
+				emit stateChanged( m_state );
 			}
 			else if( argc > 0 )
 			{
 				m_state = AuthenticationFailed;
+				emit stateChanged( m_state );
 			}
 			else
 			{
 				// failed for an unknown reason
 				m_state = ConnectionFailed;
+				emit stateChanged( m_state );
 			}
 
 			// do not sleep when already requested to stop
@@ -691,6 +696,7 @@
 	}
 
 	m_state = Disconnected;
+	emit stateChanged( m_state );
 }
 
 
